segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +219,142 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
 
     # and return the xarray to pass into calc_dist_matrix, plus the centroid_indices to readjust it
     return sample_img_xr, centroid_indices
+
+
+def generate_two_cell_test_signal_data(size_img, radius, expression,
+                                       pattern, cell_region, cell_center):
+    """
+    This function generates the signal-level test data for a specified channel (nuclear and membrane)
+    based on a given pattern and the marker num we wish to generate data for.
+
+    Args:
+        size_img: the dimensions of the image
+        radius: the radius of the cell
+        marker_num: the marker num of the cell we're looking at
+        expression: whether the channel we're looking at is nuclear or membrane
+        pattern: the specific pattern we wish to generate signal by, for now we'll just assume it's equal
+            to expression (nuclear or membrane)
+        cell_region: the region covered by the cell we're processing
+        cell_center: the center of the cell we're processing
+
+    Returns:
+        channel_signal: a NumPy array with the generated signal according to the parameters provided
+    """
+    cell_center_x = cell_center[0]
+    cell_center_y = cell_center[1]
+
+    # create the signal data array
+    signal_data = np.zeros(size_img)
+
+    # create a probability mask, which will become a set of random values based on a normal distribution
+    # either centered at a peak for nuclear, or around the edge for membrane
+    prob_mask = np.zeros(size_img)
+
+    # get the coordinates of every point in the array
+    prob_mask_rows, prob_mask_cols = np.ogrid[:size_img[0], :size_img[1]]
+
+    # generate an array of numbers valued depending on how close or far away the coordinate is
+    # from the center or the membrane
+    # note that we multiply by -random() for the membrane case because the values
+    # will initially be negative and it will help to make the values consistent
+    # also, we need to center the distribution around the center for nuclear
+    # but around an edge value for membrane
+    if expression == "nuclear":
+        prob_mask = size_img[0] + size_img[1] - np.maximum(np.abs(prob_mask_rows - cell_center_x),
+                                                           np.abs(prob_mask_cols - cell_center_y)) * random()
+        center_value = prob_mask[cell_center_x, cell_center_y]
+        prob_mask = norm.pdf(prob_mask, center_value)
+    else:
+        prob_mask = np.maximum(np.abs(prob_mask_rows - cell_center_x),
+                               np.abs(prob_mask_cols - cell_center_y)) - (size_img[0] + size_img[1]) * -random()
+        edge_value = prob_mask[cell_center_x, cell_center_y - radius]
+        prob_mask = norm.pdf(prob_mask, edge_value)
+
+    # extract only the coordinates defined by the cell region and set the prob mask accordingly
+    signal_data[cell_region[0], cell_region[1]] = prob_mask[cell_region[0], cell_region[1]]
+
+    return signal_data
```

Comment:
I think the easiest, most consistent way to do this, is to generate an ellipse w/ random values and corrupt it a little bit (something like randomly shifting every pixel value up/down/left/right).
```
# generate 1px thick ellipse coords (_perimeter for membrane, just ellipse for nuclear)
xx, yy = skimage.draw.ellipse_perimeter(cx, cy, rx, ry, rotation=angle)

# generate corruption of coordinates
xx_cor = xx + np.around(4*(np.random.rand(*xx.shape)-0.5)
yy_cor = yy + np.around(4*(np.random.rand(*yy.shape)-0.5)

# set signal at ellipse+corruption values (not necessarily 1)
signal_data = np.zeros(size_img)
signal_data[xx, yy] = 1
signal_data[xx_cor, yy_cor] = 1

return signal_data
```
Here's an example of the output of something like this:
![example](https://user-images.githubusercontent.com/36260163/89069584-48196d80-d328-11ea-824e-f4fde0956ed0.png)

Reply:
I agree, I think this will be simpler. It also addresses the issue you brought up above @alex-l-kong, which is that we aren't necessarily more confident about values in the very center of the nucleus than at the edge. 

Here's how this might look. Generate a ellipse of radius r. This is the nucleus. Assign all pixels within the nucelus to a constant value. Then, draw three perimeters of expanding radii, each of which is subsampled or jittered slightly more than the last. 

We have a lot of information that we can leverage in terms of what the signal looks like, whereas if you construct your distribution from scratch with a PDF you lose that.

Reply:
Thanks, guys for the suggestions! I'll see to implementing something similar (and simpler!).